### DISCUSSION TITLE
Deploying an api with ifcopenshell on a linux server (render.com)

### DISCUSSION THREAD

**Discussion by Alex-Schlachter27** [IfcOpenShell/IfcOpenShell#3859] [Q&A]
Hi :)

I'm currently exploring some features with ifcopenshell.
I developed an API that exposes some functions, so I can use it in a web application. 

Locally (on windows), the API works well, but when I deploy it (with render.com), i get the error: ImportError: IfcOpenShell not built for 'linux/64bit/python3.10'.

It seams like in the deployment process, ifcopenshell is installed via pip, and it uses the wrong version of ifcopenshell (not linux at least). Is it possible to specify the version in the requirement.txt somehow?

Then I also tried to add the ifcopenshell prebuilt package for linux in the site-packages of my project and add it to the github repository. But then it exceed githubs max file size. 

Any suggestions how this can be done? :)

**Comment by Moult** [comment]
What exactly is your web stack's platform? You can specify the --platform tag for pip, but if it's Linux and it isn't possible, it may simply be that you have too old of a glibc for example.

**Reply by Alex-Schlachter27** [reply]
Thanks for the quick reply! :) Really appreciate that!

I made a very simple fastAPI and tried using render.com for hosting, and that runs on linux with python 3.10, as I understand. But I'm quite new with hosting python apis. 
Locally I'm using the windows version of ifcopenshell in a seperate environment that does not get pushed to github (from where I deploy the api with render).
In the requirement.txt, I haven't specified a specific ifcopenshell version. And it seems the problem is that when the deployed version of the app is installing the dependencies from the requirement.txt, ifcopenshell gets installed with the windows version and not the linux version. Not sure how to get the propert linux-version of ifcopenshell then into the deployed api. 

Not sure glibc render is using. But it is a quite new service, so I don't think it is outdated. What are the requirements here from ifcopenshell?

**Reply by Moult** [reply]
You can use the `--platform` argument for pip like `pip --platform manylinux_2_31_x86_64 -r requirements.txt` alternatively you can specify the platform in the requirements.txt itself. See:

 - https://stackoverflow.com/questions/29222269/is-there-a-way-to-have-a-conditional-requirements-txt-file-for-my-python-applica
 - https://stackoverflow.com/questions/7225900/how-can-i-install-packages-using-pip-according-to-the-requirements-txt-file-from
 - https://stackoverflow.com/questions/49672621/what-are-the-valid-values-for-platform-abi-and-implementation-for-pip-do
 - https://pypi.org/project/ifcopenshell/#files

**Reply by Alex-Schlachter27** [reply]
Thanks! I will check it out :)
